src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/config/UserVersionFilter.kt

```diff
@@ -0,0 +1,47 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.config
+
+import org.springframework.stereotype.Component
+import org.springframework.web.filter.OncePerRequestFilter
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.ServiceName
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.UserRole
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.RequestContextService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.SentryService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.UserService
+import java.io.IOException
+import javax.annotation.PostConstruct
+import javax.servlet.FilterChain
+import javax.servlet.ServletException
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
+
+@Component
+class UserVersionFilter(
+  val sentryService: SentryService,
+  private val userService: UserService,
+  private val requestContextService: RequestContextService,
+) : OncePerRequestFilter() {
+
+  @PostConstruct
+  fun filterStartup() {
+    sentryService.captureErrorMessage("User Version filtering is enabled.")
+  }
+
+  @Throws(ServletException::class, IOException::class)
+  override fun doFilterInternal(request: HttpServletRequest, response: HttpServletResponse, filterChain: FilterChain) {
+    when (requestContextService.getServiceForRequest()) {
+      ServiceName.approvedPremises -> {
+        addUserVersionHeader(response)
+        filterChain.doFilter(request, response)
+      }
+      else -> {
+        return filterChain.doFilter(request, response)
+      }
+    }
+  }
+
+  private fun addUserVersionHeader(response: HttpServletResponse) {
+    val user = userService.getUserForRequestOrNull()
+    val userVersion = user?.let { UserRole.getVersionHashCode(it) }
+    response.setHeader("X-CAS-User-Version", userVersion.toString())
```

Comment:
i think that if the user version can't be determined (is null), we shouldn't be setting the header at all?